src/tests/util/test_app.rs

```diff
@@ -388,6 +392,17 @@ impl TestAppBuilder {
         self
     }
 
+    /// Add a new OIDC keystore to the application
+    pub fn with_oidc_keystore(
+        mut self,
+        issuer_url: impl Into<String>,
+        keystore: MockOidcKeyStore,
+    ) -> Self {
+        self.oidc_key_stores
+            .insert(issuer_url.into(), Box::new(keystore));
+        self
+    }
```

Comment:
by default, all tests start with no Trusted Publishing providers enabled, but this fn can be used to explicitly add them (with `MockOidcKeyStore` implementations).